src/vm/win32threadpool.h

```diff
@@ -372,8 +372,16 @@ class ThreadpoolMgr
 
         } counts;
 
+        // padding to ensure we get our own cache line
+        BYTE padding[64];
+
         Counts GetCleanCounts()
         {
+#ifdef _WIN64
+            // VolatileLoad x64 bit read is atomic
+            return DangerousGetDirtyCounts();
```

Comment:
There are several places in win32threadpool.cpp where after GetCleanCounts(), the counts are tested before any change is made using compare-exchange. That suggests to me that GetCleanCounts() is also intended to read a recent value of the counts, such that those tests are more representative of the current state of counts. So I think there should be a memory barrier before the load as well. It probably won't make any difference though. The memory barrier after the load is suspect, but people I have talked to claim that in cases where a volatile load is done before a compare-exchange loop, the memory barrier after the load may be necessary to prevent incorrect compiler optimizations that move the memory load and write to the local into the loop.